src/core/dataProviders/dataProvider.js

```diff
@@ -261,9 +271,23 @@ const convertHTTPResponse = (response, type, resource, params) => {
 
 const dataProvider = (type, resource, params) => {
     let { url, options } = convertDataRequestToHTTP(type, resource, params);
-    return fetch(url, options).then(response => response.json())
-        .then(res => convertHTTPResponse(res, type, resource, params))
-        .catch(err => console.log('Error: ', err));
+    let responseInfo = {};
+    return fetch(url, options).then(response => {
+        responseInfo.status = get(response, 'status', null);
+        return response.json();
+    })
+        .then(res => {
+            if (responseInfo.status !== 200) {
+                responseInfo.errorMessage = get(res, 'error', null);
```

Comment:
what's the use of setting the `errorMessage` in the `responseInfo` since you never return the `responseInfo`?

Reply:
I need both in error message and status. But I can't pass object to HttpError:
```
if (responseInfo.status !== 200) {
   responseInfo.errorMessage = get(res, 'error', null);
   throw new HttpError({
        status: responseInfo.status,
        message: responseInfo.errorMessage
    });
}
```
I can pass only string. For this reason I united status and message in one row via "|":
```
if (responseInfo.status !== 200) {
    responseInfo.errorMessage = get(res, 'error', null);
    let errorString = responseInfo.status + '|' + responseInfo.errorMessage;
    throw new HttpError(errorString);
}
```
In sagas I split string via "|" to get status and message.

Reply:
I understand, but it seems weird to set the error message value in an object rather than in a variable - especially if the shape of the object doesn't matter.

Reply:
Fixed. I created a string right away, without object.